src/walk.rs

```diff
@@ -436,6 +436,15 @@ fn spawn_senders(
                 return ignore::WalkState::Quit;
             }
 
+            // when pruning, don't descend into matching dirs
+            if config.prune {
+                if let Some(ref entry_type) = entry.file_type() {
+                    if entry_type.is_dir() {
```

Comment:
I believe these three nested `if`s could be written as:
```rust
if config.prune && entry.file_type().map_or(false, |e| e.is_dir()) {
  // ...
}
```

Reply:
Yes, that works. In diff2 I also added a check to see if there is a pattern. Because if no pattern is specified then (my impression) is that '--prune' should be ignored. (Or alternatively, is there a way to require pattern if --prune is specified during the arg checking?